Add post-process labels stage (#4)
pmhalvor/whale-speech: docs/ladr/LADR_0005_persist_intermediate_outputs.md

# Persisting intermediate stage outputs

> This doc discusses storing outputs from each stage or keeping stateless until the end, and where to do each in our pipeline. 

Author: ```suggestion
This doc discusses storing outputs from each stage or keep them stateless, and where to do each in our pipeline. 
```

Some stages (like geo-search) require storing outputs, since the (unaltered) Happywhale API currently writes found encounters to file, and does not return a df.  
<!-- TODO open an PR in happywhale to return df instead of saving to file -->
Other stages like audio retrival may make sense to keep stateless to avoid storage costs on our end. 
For debugging purposes, storing these data makes sense, since it speeds up iteration time. 
But a productionized pipeline might only run once per geofile-date, and not need to store intermediate outputs.

Another point to consider is that data-usage agreements with the audio proivders.

> The audio from MBARI is ok to download, but what happens when we scale to include other hydrophone sources? 

Author: ```suggestion
The audio from MBARI is ok to download, but we should plan for potentially scaling to include other hydrophone sources.
```
